components/sup/src/supervisor.rs

```diff
@@ -104,7 +104,7 @@ pub struct Supervisor {
     pub package: Arc<RwLock<PackageInstall>>,
     pub preamble: String,
     pub state: ProcessState,
-    pub state_entered: SteadyTime,
+    pub state_entered: Timespec,
```

Comment:
From my understanding you had to make this swap because of serialization issues. It's been a while since we've had to revisit this module so this hasn't come up before, but the crate that we're using for time is actually old and deprecated which is why you don't have the support you need in it for serde.

[chrono](https://github.com/chronotope/chrono) replaced the [time](https://github.com/rust-lang-deprecated/time) crate and it does have support for serialization using serde. You might want to consider using that here instead or we could save that for a future refactor!